File: hackerrank/30 days of code/day 28, regex, patterns, and intro to databases/regex_patterns_and_intro_to_databases.rb
# https://www.hackerrank.com/challenges/30-regex-patterns/problem?isFullScreen=true

def main
  n = gets.to_i
  names = collect_gmail_users(n)
  names.sort!
  puts names
end

def collect_gmail_users(n)
  n.times.map do
    name, email = gets.split
    name if email.end_with?('@gmail.com')
  end.compact
end

main

File: hackerrank/30 days of code/day 28, regex, patterns, and intro to databases/regex_patterns_and_intro_to_databases.cs
// https://www.hackerrank.com/challenges/30-regex-patterns/problem?isFullScreen=true

public class Solution
{
    public static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        List<string> names = CollectGmailUsers(n);
        names.Sort();
        names.ForEach(Console.WriteLine);
    }

    private static List<string> CollectGmailUsers(int n)
    {
        var names = new List<string>();

        while (n-- > 0)
        {
            string[] line = Console.ReadLine().Split();
            string name = line[0];
            string email = line[1];
            if (email.EndsWith("@gmail.com"))
                names.Add(name);
        }

        return names;
    }
}
